client/src/game/input/keyboard.ts

```diff
@@ -116,9 +160,24 @@ export function onKeyDown(event: KeyboardEvent): void {
             event.stopPropagation();
             gameStore.toggleUI();
         } else if (event.key === "0" && event.ctrlKey) {
-            // Ctrl-0 - Re-center/reset the viewport
-            gameStore.setPanX(0);
-            gameStore.setPanY(0);
+            // Ctrl-0 or numpad 0 - Re-center/reset the viewport
+            gameManager.setCenterPosition(new GlobalPoint(0, 0));
+            sendClientLocationOptions();
+            layerManager.invalidateAllFloors();
+        } else if (event.keyCode == 101) {
```

Comment:
for individual keys it's better to use `event.key` or `event.code`, `event.keycode` is deprecated but slightly easier to work with when using ranges like 37->40.  You can use something like https://keycode.info to find the correct strings.
In the case of numpad it's probably recommended to use event.code as that differentiates between number row and numpad. (you could also use event.location to differentiate)